Update GitHub Actions docs (#1761)
pulumi/docs: content/docs/guides/continuous-delivery/github-actions.md

- /docs/console/continuous-delivery/github-actions/
---

Pulumi's [GitHub Actions](https://developer.github.com/actions) help you deploy apps and
infrastructure to your cloud of choice, using nothing but code in your favorite language
and GitHub. This includes previewing, validating, and collaborating on proposed
deployments in the context of Pull Requests, and triggering deployments or promotions
between different environments by merging or directly committing changes.

Let's see how to get started -- it's easy!

> **Note**: GitHub Actions is currently in public beta. Please [register for
> access](https://github.com/features/actions) to start using the feature. Until your
> GitHub organization has been approved for the beta, the **Actions** tab won't show in

> > your repos, and the Actions-related files described below will not be recognized by

Reviewer: ```suggestion
> your repos, and the Actions-related files described in this guide will not be recognized by
```

> GitHub.

# Pre-Requisites

Before proceeding, you'll need to [Sign Up for Pulumi](https://app.pulumi.com/) (if you
haven't already). This guide also assumes you've reviewed the [GitHub Actions
documentation](https://help.github.com/en/categories/automating-your-workflow-with-github-actions)
and are generally familiar with its concepts and syntax.

For your workflow to do anything interesting, you'll want to create a new Pulumi project
for it. There are three ways to do this:

1. [Clone an existing Pulumi example](https://github.com/pulumi/examples)
2. [Use the New Project wizard](https://app.pulumi.com/site/new-project)
3. [Download the CLI]({{< relref "/docs/get-started/install" >}}) and run `pulumi new` to
   select a template

# Creating a Workflow

Although the full power of the Pulumi CLI is available to use with GitHub Actions, we
recommend starting with our standard workflow, which consists of two workflow files, each
triggered by common GitHub events:

1. **Pulumi Preview** runs `pulumi preview` in response to a Pull Request, showing what

>    would happen if the PR were merged into the target branch.

Reviewer: Maybe change to "showing a preview of the changes to the target branch when the PR gets merged"?

Let's get started by adding these two new workflow files to the GitHub repository
containing your Pulumi project.

### The pull_request Workflow File

Add a new file to your Pulumi project repository at `.github/workflows/pull_request.yml`
containing the following workflow definition, which instructs GitHub Actions to run
`pulumi preview` in response to all `pull_request` events:

```yaml
name: Pulumi
on:
  - pull_request
jobs:
  preview:
    name: Preview
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
        with:
          fetch-depth: 1
      - run: scripts/build
      - uses: docker://pulumi/actions
        with:
          args: preview
        env:
          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
          PULUMI_CI: pr

```

Note that by using the `pulumi/actions` Docker image, the workflow will automatically
download and use the latest version of Pulumi. If you prefer to use a specific version of
Pulumi, you can replace all occurrences of `uses: "docker://pulumi/actions"` in this and

> other workflow files with a container  reference that appends the desired version as a tag ---

Reviewer: ```suggestion
other workflow files with a container reference that appends the desired version as a tag ---
```

> e.g., `uses: "docker://pulumi/actions:v1.0.0"`.

Reviewer: ```suggestion
for example, `uses: "docker://pulumi/actions:v1.0.0"`.
```
